src/storage/statistics/column_statistics.cpp

```diff
@@ -64,4 +66,18 @@ shared_ptr<ColumnStatistics> ColumnStatistics::Deserialize(Deserializer &source,
 	return make_shared<ColumnStatistics>(stats.Copy(), std::move(distinct_stats));
 }
 
+void ColumnStatistics::FormatSerialize(FormatSerializer &serializer) const {
+	serializer.WriteProperty(100, "statistics", stats);
+	serializer.WritePropertyWithDefault(101, "distinct", distinct_stats, unique_ptr<DistinctStatistics>());
+}
+
+shared_ptr<ColumnStatistics> ColumnStatistics::FormatDeserialize(FormatDeserializer &deserializer) {
+	// TODO: do we read this as an property or into the object itself?
+	// we have this sort of pseudo inheritance going on here which is annoying
+	auto stats = BaseStatistics::FormatDeserialize(deserializer);
```

Comment:
Can this be generated with the json as well?

Reply:
Not yet, because of the `.Copy` call in the constructor.